### ISSUE TITLE
$secure should work without a $default value

### ISSUE THREAD

**Issue by jtanguy** [FGRibreau/common-env#27]
My usecase: I want to define that an env variable is required, but do not want to set a default value.

I might do a PR later but I just wanted to document it.

Example code:
```js
const env = require('common-env/withLogger')(console);
const config = env.getOrElseAll({
    api_url: 'https://example.com/api',
    api_key: {
        $type: env.types.String,
        $secure: true,
    },
});
```
Expected output
```
$ API_KEY="mySecretPassword" node example.js
[env] API_URL was not defined, using default: https://example.com
[env] API_KEY was defined, using: ***
```

Actual output:
```
$ API_KEY="mySecretPassword" node example.js
[env] API_URL was not defined, using default: https://example.com
[env] API_KEY was defined, using: mySecretPassword
```

**Comment by FGRibreau** [comment]
Indeed, that's a valid usecase, I will accept a PR for it @jtanguy :)

**Comment by FGRibreau** [comment]
A quick-workaround could be :

```js
require('common-env/withLogger')(logger)
  .on('env:fallback', function(fullKeyName, $default, $secure) {
    // see https://github.com/FGRibreau/common-env/issues/27
    if ($secure && !process.env[fullKeyName]) {
      throw new Error(`${fullKeyName} environment variable must be defined`);
    }
  })
  .getOrElseAll({
...
```

**Comment by FGRibreau** [comment]
Released in v6.3.0
